tests/test_s3blobstore.py

```diff
@@ -145,5 +155,134 @@ def test_get_bucket_region(self):
         self.assertEqual(handle.get_bucket_region(self.test_us_east_1_bucket), 'us-east-1')
         self.assertNotEqual(handle.get_bucket_region(self.test_non_us_east_1_bucket), 'us-east-1')
 
+    def test_read_timeout(self):
+        read_timeout = 1
+        with contextlib.closing(ProxyConnectServer.start(2 * read_timeout)):
+            config = botocore.config.Config(
+                proxies={
+                    'http': f'{ProxyConnectServer.address}:{ProxyConnectServer.shared_info["port"]}',
+                    'https': f'{ProxyConnectServer.address}:{ProxyConnectServer.shared_info["port"]}',
+                },
+                read_timeout=read_timeout,
+                retries={'max_attempts': 0}
+            )
+            s3_client = boto3.client('s3', config=config)
+            handle = S3BlobStore(s3_client)
+
+            # Make sure we actually raise a ReadError
+            with self.assertRaises(ReadTimeout):
+                s3_client.put_object(
+                    Bucket=self.test_bucket,
+                    Key='fake_key',
+                    Body=os.urandom(1000)
+                )
+
+            # Make sure we correctly respond to a ReadError
+            with self.assertRaises(BlobStoreTimeoutError):
+                handle.upload_file_handle(
+                    self.test_bucket,
+                    'fake_key',
+                    io.BytesIO(os.urandom(1000))
+                )
+
+    def test_connect_timeout(self):
+        """
+        Ensure that we handle botocore ConnectTimeouts
+        """
+        s3_client = boto3.client(
+            's3',
+            config=botocore.config.Config(
+                retries={'max_attempts': 0}
+            )
+        )
+
+        # Point boto to an unresponsive host
+        s3_client._endpoint = botocore.endpoint.Endpoint(
+            'https://www.chanzuckerberg.com:3000',
+            '',
+            s3_client._endpoint._event_emitter
+        )
+        handle = S3BlobStore(s3_client)
+
+        with self.assertRaises(BlobStoreTimeoutError):
+            handle.upload_file_handle(
+                self.test_bucket,
+                'fake_key',
+                io.BytesIO(os.urandom(1000))
+            )
+
+
+def unused_tcp_port():
+    with contextlib.closing(socket.socket()) as sock:
+        sock.bind((ProxyConnectServer.address, 0))
+        return sock.getsockname()[1]
+
+class ProxyConnectServer(HTTPServer):
+    address = '127.0.0.1'
```

Comment:
double quotes for strings that are not keys.  this should be applied throughout this PR.

Reply:
Quotable nits? Fixed.